### PR TITLE
OpenTelemetry on all endpoints

### PR DESCRIPTION
**Pull request by stefanmcshane**
- Bump chi to chi/v5 per https://github.com/go-chi/chi/blob/master/CHANGELOG.md#v500-2021-02-27
- Added `riandyrn/otelchi` to all endpoints
- Protected `telemetry.Shutdown()` in cases of errors

### CODE REVIEW

File: api/server/handlers/stacks/list_porter_app_events.go [hunk 1]
```diff
@@ -38,20 +38,28 @@ func NewPorterAppEventListHandler(
...
 	cluster, _ := ctx.Value(types.ClusterScope).(*models.Cluster)
+	telemetry.WithAttributes(span,
+		telemetry.AttributeKV{Key: "cluster-id", Value: int(cluster.ID)},
+		telemetry.AttributeKV{Key: "project-id", Value: int(cluster.ProjectID)},
+	)
 
 	stackName, reqErr := requestutils.GetURLParamString(r, types.URLParamStackName)
 	if reqErr != nil {
-		p.HandleAPIError(w, r, apierrors.NewErrPassThroughToClient(reqErr, http.StatusBadRequest))
+		e := telemetry.Error(ctx, span, nil, "error parsing stack name from url")
```
Comment (d-g-town) [inline api/server/handlers/stacks/list_porter_app_events.go:52]: do we want to pass in `reqErr`?

Comment (d-g-town) [reply api/server/handlers/stacks/list_porter_app_events.go:52]: Same question for `err` in the other telemetry.Error calls below

**Review by d-g-town — APPROVED** [review]

File: api/server/handlers/stacks/list_porter_app_events.go (hunk 1 above)
Comment (stefanmcshane) [reply api/server/handlers/stacks/list_porter_app_events.go:52]: good catch

Comment (stefanmcshane) [reply api/server/handlers/stacks/list_porter_app_events.go:52]: reqErr is a specific type, so it's only valid in this one case. The rest are all of type `error`

**Pull request merged**